WordPress/Classes/Categories/NSString+Helpers.m

```diff
@@ -188,4 +188,22 @@ - (NSArray *)tokenize
     return tokens;
 }
 
+- (BOOL)isWordPressComPath
+{
```

Comment:
At the risk of piling on, and also cos @astralbodies's regex hate got me to thinking ;), maybe using `NSURLComponents` is enough.  For instance, `[components.scheme hasPrefix:@"http"]` is probably sufficient for validating the protocol and `[components.host hasSuffix:@wordpress.com"]` looks like its sufficient for identifying a wpcom path.